src/audio/pipeline.c

```diff
@@ -320,7 +320,6 @@ static int pipeline_comp_hw_params(struct comp_dev *current,
 				   struct pipeline_walk_context *ctx, int dir)
 {
 	struct pipeline_data *ppl_data = ctx->comp_data;
-	int ret = 0;
```

Comment:
Keep this here, we are following MISRA rules as close as we can for safety critical in the future.

Reply:
This is the first time I hear about it. Do we have the standard somewhere? Do we really check for it and reinforce it? Is the current code base really following it to a significant degree? Are you referring only to this specific initialisation or to all of them?

Reply:
We do check for it, but it's manual and not automatic atm. The code should be following this in all places. If not we can create fixes.

Reply:
@lgirdwood sorry, II looked through MISRA 2004 requirements and I don't see anything like this. Could you point me out to a specific clause there?

Reply:
This came from safety critical folks, so lets do it.